sparqlbuilder/src/main/java/org/eclipse/rdf4j/sparqlbuilder/rdf/Rdf.java

```diff
@@ -21,42 +27,60 @@
 import org.eclipse.rdf4j.sparqlbuilder.rdf.RdfLiteral.StringLiteral;
 
 /**
- * A class with static methods to create basic RDF objects
+ * A class with static methods to create basic {@link SparqlBuilder} RDF objects from either string values or
+ * RDF4J Model objects.
  */
 public class Rdf {
+
 	// not sure if other protocols are generally used in RDF iri's?
-	private static final Set<String> IRI_PROTOCOLS = Stream.of("http://", "https://", "mailto:").collect(Collectors.toSet());
+	private static final Set<String> IRI_PROTOCOLS = Stream.of("http://", "https://", "mailto:").collect(
```

Comment:
I've also used "tel:" actually (in vcard stuff, but admittedly very rare), but `urn:` is not uncommon (especially 
 `urn:uuid:`)

Reply:
Yes, this is a problem in the current SparqlBuilder codebase. We should look into letting it use RDF4J's IRI validation codes instead of this ad-hoc string parsing.  

I'm currently in rather a hurry, sorry, if you have time could you log this as a new bug? It's not something I'll address in this PR but we don't want to forget about it.

Reply:
Ok, I've created  #1270